open-metadata-conformance-suite/open-metadata-conformance-suite-api/src/main/java/org/odpi/openmetadata/conformance/ffdc/ConformanceSuiteErrorCode.java

```diff
@@ -234,6 +234,18 @@ public String getUnformattedErrorMessage()
         return errorMessage;
     }
 
+    /**
+     * Returns the string but with any special storage characters changes to underscores
+     *
+     * @return String
+     */
+    private String sanatiseLogString( String str ) {
+        /*
+        quietly change any characters involved in storage hacks to a harmless underscore.
+         */
+        return str.replaceAll("[\n\r\t]", "_");
```

Comment:
I wonder if we should consider using the owasp logging plugin together with a logging configuration-> https://cheatsheetseries.owasp.org/cheatsheets/Injection_Prevention_in_Java_Cheat_Sheet.html

This could give us the best, quickest coverage - though wouldn't directly avoid the scanners reporting an issue in the code - and then we would need to deal with that!

Given that, I think the replacement of chars as you've done is a sound solution - and doesn't prevent us ALSO doing the owasp approach. it's something though they we would likely need across many components, which may need refactoring. But it's a step forward


May be useful to also check str isn't null? (or is this a no given the scan?)

Reply:
Ok - sounds good. Any improvement is fine, I was just wondering if - as you say - it was worth considering the bigger issue (async, diff pr)!. Will close as you are handling it